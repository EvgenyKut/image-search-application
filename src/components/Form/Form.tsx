import React, { SyntheticEvent } from 'react';
import { useKeyListener } from '../../hooks';

import styles from './form.module.css';

type FormProps = {
  onSubmit: (e: SyntheticEvent) => void;
  children: React.ReactNode;
};

const Form: React.FC<FormProps> = ({ children, onSubmit }) => {
  useKeyListener('Enter', onSubmit);
  return (
    <div onSubmit={onSubmit} className={styles.wrapper} data-testid="form">
      {children}
    </div>
  );
};

export default Form;
