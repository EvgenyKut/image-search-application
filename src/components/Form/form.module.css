.wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
}

body[data-theme='dark'] .wrapper {
  background-color: lightblue !important;
  filter: brightness(1.1);
  border-radius: 8px;
}
