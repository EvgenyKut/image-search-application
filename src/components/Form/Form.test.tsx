import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Form from './index';

describe('Form-component render test', () => {
  const testContent = 'Test';
  const mockCallback = jest.fn();
  it('should have snapshot', () => {
    const tree = renderer
      .create(
        <Form onSubmit={mockCallback}>
          <div>{testContent}</div>
        </Form>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Form test cases', () => {
    render(
      <Form onSubmit={mockCallback}>
        <div>{testContent}</div>
      </Form>,
    );
    const textField = screen.getByTestId('form') as HTMLInputElement;
    expect(textField).toBeInTheDocument();
    expect(textField).toBeVisible();
    expect(textField.textContent).toBe(testContent);
  });
});
