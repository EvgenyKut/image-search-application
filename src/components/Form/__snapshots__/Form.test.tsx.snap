// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form-component render test should have snapshot 1`] = `
<div
  className="wrapper"
  data-testid="form"
  onSubmit={[MockFunction]}
>
  <div>
    Test
  </div>
</div>
`;
